### ISSUE TITLE
Coins on stealth address can't be moved...

### ISSUE THREAD

**Issue by Aergon** [vergecurrency/VERGE#579]
## Context

I did made a new wallet with several address inside, some of them are stealth address, i send 10 coins to regular address and 20 coins to stealth address from another wallet, after that i try move the 30 coins outside wallet and was imposible, so i begin to try make smaller transactions and i was able to getout 10 coins, but 20 coins can't be moved, think are the 20 coins in stealth address, tried several things, wallet is encripted, so i did unlock and try again without success,

After that i tried get private key for stealth address but can't neither.

## Prerequisites

* [x ] Are you running the latest version?

yes

## User Story:

* "I'm submitting a ... "
  - [x] bug report
  
I am using windows 10 64bits and Wallet-qt version: v4.0.1.0-794660a-funky-beta

## Expected Behavior

Coin get transfered to another wallet

and / or 

Get private key from stealth address for try make a new wallet in Electrum

## Current Behavior

The Wallet-qt program show me this message when try make the transaction:
" Error. Fallo en la creacion de la transaccion."

and / or

When ask for private key on console program fail in give it.

## Screenshots

![image](https://user-images.githubusercontent.com/35868780/35473057-8d3b71c2-037a-11e8-9e5c-c2b16ec29677.png)

![image](https://user-images.githubusercontent.com/35868780/35473064-aa3cd6bc-037a-11e8-93f4-1c0b3d7ebdc3.png)

![image](https://user-images.githubusercontent.com/35868780/35473086-f7fa9b1e-037a-11e8-8eb7-60cb765e4c0c.png)

![image](https://user-images.githubusercontent.com/35868780/35474157-9dd46696-038a-11e8-8ee3-cd75a5afc77f.png)

![image](https://user-images.githubusercontent.com/35868780/35474459-cdcfbf0e-038e-11e8-849a-f262e55bfa94.png)

![image](https://user-images.githubusercontent.com/35868780/35474157-9dd46696-038a-11e8-8ee3-cd75a5afc77f.png)

![image](https://user-images.githubusercontent.com/35868780/35491739-b179c4ba-04a8-11e8-8093-c253ff78d7e3.png)

## Possible Solution

Upgrade the wallet program for fix some minor bugs.

## Steps to Reproduce (for bugs)

* Steps :
1. Create new wallet
2. Create 2 regular address and 2 stealth address
3. Send coins to new regular address and new stealth address
4. Try send coins outside the wallet to another wallet
5. Try recover private key from stealth address

## Your Environment

Verge Wallet: Windows Qt Version: v4.0.1.0-794660a-funky-beta
Windows 10 64bits
i7 4700HQ 2.40 Ghz
32mb Ram
GPU: Nvidia 770m

**Comment by Technophant** [comment]
Thanks for the detailed report.  I was playing around with this too today by sending myself a stealth transaction.  I noticed that is wasn't in the transaction history. Not sure if this a feature or a bug.  I did just now trying sending out the incoming 9xvg in 2 transactions and both were received by electrum. 

Your coins are in the Dmcdc address I think or a change address. I'm not sure how stealth addressing works yet.  It goes into a new D address everytime your receive I was told.  Go to the debug windows and type "listtransactions" and you can start looking up the tx's on verge-blockchain.info or post it here so we can help you decode it. You can also look at debug.log and see if there's any recent entries that are relevant. 

I'm hoping your kids won't go hungry unless you get those 20xvg back. ;-)

**Comment by Aergon** [comment]
Thanks for comments Technophant, i will try the command you said and look log and will post results

**Comment by Aergon** [comment]
Now can be a few dollars but can be a lot more in few months, we will see :P , i like this project and have nice specifications, will be a greater community every day.

**Comment by Aergon** [comment]
The worst part is i can't retrieve the private key of stealth address . Something is wrong there, is like wasn't mine.

**Comment by Aergon** [comment]
![image](https://user-images.githubusercontent.com/35868780/35568310-cf3f555c-05c8-11e8-976f-58b7bf59aa3d.png)

no more transactions when i execute the command: "listtransactions"

**Comment by Aergon** [comment]
Always after i try and fail on make a transaction Verge-Qt goes to 100 CPU use resources and need close it

**Comment by justinvforvendetta** [comment]
you can't retrieve the privkey to the DMCv address?

**Comment by jrsikors** [comment]
This probably won't fix the stealth address thing , but do this:

Step 1) Lock your wallet
Step 2) Go to debug console again
Step 3) If your passphrase is "Aergon rocks" ... type this:
walletpassphrase "Aergon rocks" 300
This unlocks your wallet for 300 seconds
Step 4) Do a dumpprivkey on that address again

Using the unlock through the GUI menu doesn't seem to unlock your wallet and be able to dumpprivkey.   If you want to dumpprivkey you need to do the walletpassphrase command in console to unlock it first

**Comment by Aergon** [comment]
Thanks for the comment Jrsikors, did already for be able to use dumpprivkey, but give me the error wich show in screenshot in first post.

No, Justinvforvendetta, can't retrieve the private key of any address related with stealth one, forgive my lack of knowledge, but i don't know what you mean with DMCv address, like screenshot show, i try both address, stealth and related one, not working on any of this. But dumpprivkey work in the other address i keep on same wallet. Pretty sure i did something wrong when create address or something but can't figure out what is.

Still learning about XVG, and testing things, will be great if can point me out a good source of knowledge about full mechanic of Verge, i am confused like private keys save the coins because i made wallets and after a few movements in and out and export private key to electrum xvg, my balance in wallets are different against block chain explorer, i can develop this in another thread if you wish.

My offer remains for developers, i can donate the wallet for research if you point me where i need send it. I don't know what else do with it  :) 

Thanks for support and keep the good work.

**Comment by jrsikors** [comment]
He just meant ... you should be able to do a:
dumpprivkey **_DMCv_**DCyCsuwX4p6WCtuQWEmXfaq4A4ujjj

He just didn't want to type out the whole address because you only included a screenshot and we can't copy/paste those :smile:

**Comment by jrsikors** [comment]
We need to fix that before you can move on.

If you open Verge, and go to "Receiving Addresses" on the left ... is DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj listed in there?

If it is -- you should be able to:
 1) Go to console
 2)  walletpassphrase "<your pass phrase>" 300  to unlock wallet
 3) type dumpprivkey DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj

If you can't even do that -- you got some weird issues going on that we need more info on to help with

**Comment by Aergon** [comment]
Thanks for reply, indeed i tried that, if you check the last screenshot on first post you can see that, but give the error: "Private key for address... is not known (Code -4)" 

Seems to be i don't own that address, can't think on another thing for try.

I did downloaded the Verge-qt version 4 several times few days ago, maybe i make the stealth address with an old version and screw it without known, and now even on last version is not working for something, i even tried use version 3 and give me another error, but can't retrieve address private key neither.

I did try again your recomendation, and here is the new screenshot:

![image](https://user-images.githubusercontent.com/35868780/35603231-96920960-063c-11e8-8e43-a33dafd02695.png)

**Comment by jrsikors** [comment]
Justin will probably need to help, but couple more follow-up questions that might help him:
  1)  Did you create all 5 of those addresses in the verge-qt wallet?  
    1A)  Did you create them by the 'receive address icon on the left' and clicking the "New Address" button?  Or did you create them through the console?
   2)  Can you get private keys for ANY of the 5 addresses?

What is the output of "listaddressgroupings" from the console ?

**Comment by Aergon** [comment]
1. Did you create all 5 of those addresses in the verge-qt wallet?
Yes, i did, but can't remember if was last version 4 or a older one version 4.

2. Can you get private keys for ANY of the 5 addresses?
Yes, i can get private keys from at least 1 regular address, i didn't try the other one, and the other stealth one never got funds so seems to be don't have a normal address related to it for try retrieve private key.

**Comment by Aergon** [comment]
What is the output of "listaddressgroupings" from the console ?

![image](https://user-images.githubusercontent.com/35868780/35661829-15dc5ff8-0715-11e8-8387-eb10b0116463.png)

**Comment by Aergon** [comment]
I just test get private key from both regular address, both worked, just can't from stealth related one.

**Comment by Aergon** [comment]
![image](https://user-images.githubusercontent.com/35868780/35662076-61dcb5b4-0716-11e8-94ce-bb8865da6b8d.png)

**Comment by Aergon** [comment]
Yesterday i was trying some different stuff, so i use a different computer, download blockchain from 29th January, download last qt version 4 wallet, and install all, copy the wallet.dat to the Verge directory and wait for everything got updated.

Try 2 more things, use -debug and use -resync, after finish all, still can't transfer or get private key from stealth address, hope this help you with the troubleshooting.

Greets.

**Comment by Aergon** [comment]
Sorry i didn't answered it yet:

1A) Did you create them by the 'receive address icon on the left' and clicking the "New Address" button? Or did you create them through the console?

I create all addres from "New Address" button from icon on the left, just when i make stealth ones i click the stealth box and get checked.

**Comment by Aergon** [comment]
Btw, after delete Verge folder, and download blockchain again and let wallet make everything again, my Qt wallet don't eat all my resources when transaction fail, now have a normal behavior.

I have at least 4 different verge wallets working on same computer, just alternate it inside Verge folder for use. the other 3 are working well but none of this have stealth address inside. Just FYI.

**Comment by justinvforvendetta** [comment]
stealth doesnt dump a private key, because stealth is actually sent to a random pubkey. whichever coins came in from your stealth, came in from a public address, that you can definitely retrieve the key to. the only thing is, if you receive coins to that stealth addr again, it will be a different public address, and so on.

**Comment by justinvforvendetta** [comment]
also, any wallet other than the current one, is no longer compatible with the current blockchain, and will not allow the data directory you use, to function on the new chain, as you will be at a different height when using the new wallet.

**Comment by Aergon** [comment]
Question: if i did stealth address with old wallets will not longer work even if i use last Windows-QT wallet version?

**Comment by Aergon** [comment]
Another Question: There is another way for recover the coins inside this wallet?

**Comment by jrsikors** [comment]
I'm pretty sure you can 'recover' funds on a stealth address that was created with old Windows-QT.

I've noticed people use the terminology 'wallets' and 'qt app' in a way that can be confusing.

A) Your **_wallet_** with your addresses/funds is your wallet.dat.
B) Your **_QT-wallet UI_** is essentially just a UI to access your wallet (A)

Justin was saying that the last QT UI (B) is not compatible and you will definitely have issues sending / receiving.

If you created (A) in the old version of (B), you should still be fine.  (A) can be copied from (B) to (B) to (B).

If you are using new (B), your (A), regardless of what (B) you created it in, should still be fine.

Is that not the case?  I noticed in your screenshots you have 20XVG.  Can you not send those 20 XVG somewhere else?

**Comment by jrsikors** [comment]
but actually I might see your dilemma ...
Your address has 20 XVG in it:
https://verge-blockchain.info/address/DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj

but even though it's showing in your wallet.. you can't seem to be able to dump the private key for **_DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj_** .. which is odd and I have no explanation for that.

But can you send your 20 coins... (19.9 with the .1 fee) to another address?  Do you get an error?

**Comment by Aergon** [comment]
Hi, thanks to Jrsikors and Justin for follow the thread, i know can be confusing, sorry, english is not my first language, forgive me any errors or misleading.

Thanks for last clarification.

I am using last (B) UI, but can't remember in wich version i did create the stealth address.

Quote:
"but even though it's showing in your wallet.. you can't seem to be able to dump the private key for DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj .. which is odd and I have no explanation for that.

But can you send your 20 coins... (19.9 with the .1 fee) to another address? Do you get an error?"

Indeed, i can't dump the private and can't sent coins to another address, is the source of my issue, if you check screenshots in first post, you can see the error the UI give me: "Fallo en la creacion de la transaccion" in English is something like: "Failed making the transaction" or similar.

Thanks in advance for all help and support, i am sure this will be fixed, just matter of time and some patience.

**Comment by jrsikors** [comment]
Well luckily you are smart enough not to send hundreds or thousand of dollars worth of coins without testing it... but in all honesty I think you might have lost those 20 coins.

It sounds like the coins are on:
DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj

and you can't get the private key for it... and you can't send coins from it... so it really does sound like your verge-QT doesn't think you really own the address.  Usually when that happens you are out of luck :(

Maybe @justinvforvendetta has some other ideas, but

**Comment by jrsikors** [comment]
also this is interesting too -- 
I can't find DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj on the new verge blockchain explorer
vergechain.org

**Comment by jrsikors** [comment]
I know vergechain.org is new, so maybe there are some bugs, but that's pretty weird IMO:

https://vergechain.org/address/DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj

https://verge-blockchain.info/address/DMCvDCyCsuwX4p6WCtuQWEmXfaq4A4ujjj

**Comment by justinvforvendetta** [comment]
I think vergechain is accurate in this instance

**Comment by jrsikors** [comment]
got me curious now @justinvforvendetta  ... not arguing at all, you are the king, but why would verge-blockchain.info be inaccurate?  It using outdated code referencing the 'old chain' ?

**Comment by justinvforvendetta** [comment]
my guess would be the explorer db needs to be resynced, because the daemon under it seems to be running the latest version.

**Comment by jrsikors** [comment]
block 0b0a57c7c367194c16133c9fb32ef1fa210cc32483380a654177d61f95a65f82 is different across the 2... where his transaction supposedly happened.

https://vergechain.org/block/0b0a57c7c367194c16133c9fb32ef1fa210cc32483380a654177d61f95a65f82

https://verge-blockchain.info/block/0b0a57c7c367194c16133c9fb32ef1fa210cc32483380a654177d61f95a65f82

![vergechain org](https://user-images.githubusercontent.com/31370541/35781358-c5fa0b2e-09ae-11e8-96dc-234a41389016.png)


![verge-blockchain info](https://user-images.githubusercontent.com/31370541/35781359-c90e18c8-09ae-11e8-88c7-ccbd7a0d0f72.png)

**Comment by jrsikors** [comment]
but yeah -- more of the same.  the 2nd screenshot , i highlighted in purple-ish the ones that match across the chains.  The yellow highlighted one is Jose's transaction.  

The non highlighted ones don't show on vergechain.org either.

**Comment by Aergon** [comment]
You are getting closer, thanks for all time you are investing on this Jrsikors. Don't worry about the 20 coins, but hope this thread lead on get this kind issue fixed.

Thanks Justin for help too, if you want try something else, let me know.

**Comment by Technophant** [comment]
Vergechain.org has issues, transactions on there that weren't confirmed and it seems missing ones as well.  From the console you can getblock 0b0a57c7c367194c16133c9fb32ef1fa210cc32483380a654177d61f95a65f82 and see that his transaction was included.  gettransaction d4bc4ae9d435f718250787b2148245bfe084d94ba872b54a00225495acaf51f4
 for more on it.  Issues seem to have come about after the hardfork.  Vergechain needs to be reset and rebuilt.

**Comment by topcatx** [comment]
I was able to spend the amount in the stealth address without any problem. However, I did notice a weird behavior: 
Wallet Status: 
- Public Address: 2 coins 
- Stealth Address: 3 coins. 

If I transfer 5 coins to another address both public/stealth address appear as input addresses for the transaction at verge-blockchain.info. 

This could possibly mean that a stealth address can be tied to a public address. 

Wondering if we should have a checkbox/dropdown while sending coins so that
  - Users can choose to send coins from both public/stealth addresses in the wallet
  - Users can choose to send coins only from public addresses in the wallet 
  - Users can choose to send coins only from stealth addresses in the wallet

think this will give more flexibility for the users. @justinvforvendetta thoughts ?
